gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy

```diff
@@ -203,6 +294,29 @@ class BaselineIdea extends AbstractBaselinePlugin {
         }
 
         project.logger.debug "Baseline: Configuring Checkstyle for Idea"
+
+        addCheckstyleNode(node)
+        addCheckstyleExternalDependencies(node)
+    }
+
+    private static void addCheckstyleIntellijImport(project) {
+        def checkstyle = project.plugins.findPlugin(BaselineCheckstyle)
+        if (checkstyle == null) {
+            project.logger.debug "Baseline: Skipping IDEA checkstyle configuration since baseline-checkstyle not applied"
+            return
+        }
```

Comment:
I'd flip this so it performs the action _when_ (and if) the BaselineCheckstyle plugin is applied.
In the current state, you're dependent on the order of application, so if BaselineCheckstyle is applied later, intellij won't be configured correctly.

```groovy
project.plugins.withType(BaselineCheckstyle) {
  // rest of the function
}
```

Reply:
I copied this from the existing checkstyle application. Would you recommend I flip it for both then?

Reply:
yep!

Reply:
I ended up flipping it for both `addCheckstyle` (called from within `ideaRootModel.project.ipr.withXml` so maybe that wasn't needed?) and `addCheckstyleIntellijImport` (mostly because I didn't see your edit before pushing, but I also think consistency is better - I'd even want to not have duplicate code but didn't find a nice way to do it because I'm not familiar enough with the code and how gradle configuration works yet)